src/app/static/src/tests/components/forms/dynamicForm.test.ts

```diff
@@ -70,85 +76,80 @@ describe('Dynamic form component', function () {
 
     it("does not render button if includeSubmitButton is false", () => {
         const rendered = getWrapper({includeSubmitButton: false}, shallowMount);
         expect(rendered.findAll("button").length).toBe(0);
     });
 
-    it("adds was-validated class on submit", async () => {
-        const rendered = getWrapper({}, mount);
-        rendered.find("button").trigger("click");
-        await Vue.nextTick();
-        expect(rendered.classes()).toContain("was-validated");
+    it("button is disabled while required values are missing", () => {
+        const rendered = getWrapper({}, shallowMount);
+        expect(rendered.find("button").attributes("disabled")).toBe("disabled");
     });
 
-    describe("valid submission", () => {
-
-        it("emits event with serialised form data on button submit", () => {
-            const rendered = getWrapper({formMeta: formMetaWithAllControlTypes}, mount);
-            rendered.find("[name=id_1]").setValue(10);
-            rendered.find("[name=id_3]").setValue("opt1,opt2");
-            rendered.find("button").trigger("click");
-            expect(rendered.emitted("submit")[0][0]).toStrictEqual({
-                "id_1": "10",
-                "id_2": null,
-                "id_3": ["opt1", "opt2"],
-                "id_4": null
-            });
-        });
-
-        it("emits event and returns serialised form data on programmatic submit", () => {
-            const rendered = getWrapper({formMeta: formMetaWithAllControlTypes}, mount);
-            rendered.find("[name=id_1]").setValue(10);
-            const result = (rendered.vm as any).submit();
-            const expected = {
-                data: {
-                    "id_1": "10",
-                    "id_2": null,
-                    "id_3": [],
-                    "id_4": null
-                },
-                valid: true,
-                missingValues: []
-            };
-            expect(rendered.emitted("submit")[0][0]).toStrictEqual(expected.data);
-            expect(result).toStrictEqual(expected);
-        });
+    it("button is enabled when required values are present", () => {
+        const formMeta = {
+            controlSections: [{
+                label: "Test 2",
+                controlGroups: [{
+                    label: "Group 1",
+                    controls: [
+                        {
+                            name: "id_1",
+                            type: "number",
+                            required: true,
+                            value: 10
+                        } as NumberControl,
+                        {
+                            name: "id_1",
+                            type: "number",
+                            required: false
+                        } as NumberControl
+                    ]
+                }]
+            }]
+        };
+        const rendered = getWrapper(formMeta, mount);
+        expect(rendered.find("button").attributes("disabled")).toBeUndefined();
```

Comment:
This is driving me crazy - why is this test failing??? Have run up the app and confirmed that the disabled/enabled state works as expected. But this test and the next one are failing. Any ideas?